### PR TITLE
 Adding tests, adding connection time out for feign, refactoring 

### PR DESCRIPTION
**Pull request by Yasara123**
## Automation tests
 - Unit tests 
   > Unit test

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+        eventArrived = false;
+        SiddhiMetricsDataHolder.getInstance().setMetricService(metricService);
+        SiddhiMetricsDataHolder.getInstance().setMetricManagementService(metricManagementService);
+        metricManagementService.startReporter("Console");
+    }
+
+    @AfterSuite
+    protected static void destroy() throws Exception {
+        metrics.deactivate();
+    }
```
Comment (gimantha) [inline components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java]: Add new line here

File: components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+            @Override
+            public boolean containsBufferedEvents() {
+                return true;
+            }
+        };
+       bufferedEventsTracker.registerEventBufferHolder(eventBufferHolder,"test.size");
+        AssertJUnit.assertEquals("test.size",bufferedEventsTracker.getName(eventBufferHolder));
+    }
+    @Test
```
Comment (gimantha) [inline components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java]: New line here

File: components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+            }
+        });
+
+        InputHandler inputHandler = siddhiAppRuntime.getInputHandler("cseEventStream");
+        siddhiAppRuntime.start();
+        for (int i=0; i<1;i++) {
+            inputHandler.send(new Object[]{"WSO2", 55.6f, 100});
+            inputHandler.send(new Object[]{"IBM", 75.6f, 100});
+        }
+        Thread.sleep(3010);
```
Comment (gimantha) [inline components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java]: Why exactly 3010 millisec? Can we replace this with awaitility  lib?

File: components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java (hunk 1 above)
Comment (Yasara123) [reply components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java]: Fixed in b4a2bcf444d25031775a679bad76798d70ce7d5b

File: components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java (hunk 2 above)
Comment (Yasara123) [reply components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java]: Fixed in b4a2bcf444d25031775a679bad76798d70ce7d5b

File: components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java (hunk 3 above)
Comment (Yasara123) [reply components/org.wso2.carbon.siddhi.metrics.core/src/test/java/org/wso2/carbon/siddhi/metrics/core/StatisticsTestCase.java]: Fixed in b4a2bcf444d25031775a679bad76798d70ce7d5b

**Pull request merged**